vm_ip_wait.go

```
@@ -31,10 +32,34 @@ func waitForIPAddresses(
 				return err
 			}
 			if len(result) == 0 {
+				if !hasNICs {
+					// We check if the VM has network interfaces and warn the user if that's not the case.
+					hasNICs, err = checkIfVMHasNICs(client, id, retries)
+					if err != nil {
+						// If a specific error was returned, return that, otherwise fall back on the normal
+						// EPending below.
+						return err
+					}
+				}
 				return newError(EPending, "no IP addresses reported yet")
 			}
 			return nil
 		},
 	)
 	return result, err
 }
+
+func checkIfVMHasNICs(client Client, id VMID, retries []RetryStrategy) (bool, error) {
+	networkInterfaces, err := client.ListNICs(id, retries...)
+	if err != nil {
+		return false, nil
```

why it returns nil when an error occurs?

---

Because we purposefully ignore this error since the check only runs in an error case.